numba/tests/test_typeinfer.py

```diff
@@ -142,6 +143,14 @@ def test_integer(self):
                 expected = self.int_unify[key[::-1]]
             self.assert_unify(aty, bty, getattr(types, expected))
 
+    def test_bool(self):
+        aty = types.boolean
+        for bty in types.integer_domain:
+            self.assert_unify(aty, bty, bty)
+        # Not sure about this one, but it respects transitivity
+        for cty in types.real_domain:
+            self.assert_unify(aty, cty, cty)
```

Comment:
I never knew I could type `False + 1.0`, but it does work (and returns `2.0`).   :)